### PR TITLE
Fix HC styles for Dropdown and ComboBox items to remove noisy borders

### LINKED ISSUE #14183: The dropdown, split button and combo box menus have too many lines in high contrast, making it hard to see focus or read
### Environment Information

Edge + Windows 10

### Describe the issue:

Go to the sample page for dropdown, combobox, or split button
Turn on High Contrast
Click the control

Result: Menu is a mess of noise, making it difficult to see options and the focused option
image

Expect: Should look like Contextual Menu
image

### PR DESCRIPTION
**Pull request by tomi-msft**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #14183 
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Fix HC styles for Dropdown and ComboBox items to remove noisy borders.

**Before** 
Dropdown and ComboBox menus had too many lines, making it hard to read or focus:
 **Dropdown**
image
  **ComboBox**
image

**After**
Dropdown and ComboBox menu items have borders removed
  **Dropdown**
image
  **ComboBox**
image

Main files to test are Dropdown.styles.ts and ComboBox.styles.ts. Rest are change files or snapshot updates

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts [hunk 1]
```diff
@@ -229,15 +229,15 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
...
+                  //borderColor: 'Highlight',
                   color: 'Highlight',
                 },
               },
             },
           ],
           ['&:active .' + globalClassnames.title]: [
             !disabled && rootHoverFocusActiveSelectorNeutralDarkMixin,
-            { borderColor: palette.themePrimary },
+            //{ borderColor: palette.themePrimary },
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts:240]: revert

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts [hunk 2]
```diff
@@ -198,7 +198,7 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
       fonts.medium,
       {
         color: semanticColors.menuItemText,
-        borderColor: semanticColors.focusBorder,
+        //borderColor: semanticColors.focusBorder,
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts:201]: revert

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts [hunk 3]
```diff
@@ -229,15 +229,15 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
 
               selectors: {
                 [HighContrastSelector]: {
-                  borderColor: 'Highlight',
+                  //borderColor: 'Highlight',
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts:232]: remove?

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts [hunk 4]
```diff
@@ -144,7 +144,7 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
           highContrastItemAndTitleStateMixin,
         ],
         [HighContrastSelector]: {
-          borderColor: 'Window',
+          //borderColor: 'Window',
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts:147]: remove?

File: change/@fluentui-react-next-2020-07-30-18-07-42-a11y-menu.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "prerelease",
+  "comment": "update snapshots",
+  "packageName": "@fluentui/react-next",
```
Comment (xugao) [inline change/@fluentui-react-next-2020-07-30-18-07-42-a11y-menu.json:4]: once we finalized fixes in dropdown.styles.ts , we need to make the same changes in this packge too

File: packages/experiments/src/components/Pagination/__snapshots__/Pagination.test.tsx.snap [hunk 6]
```diff
@@ -1783,7 +1767,6 @@ exports[`Pagination render comboBox Pagination correctly 1`] = `
           z-index: 1;
         }
         @media screen and (-ms-high-contrast: active){.ms-Fabric--isFocusVisible &:focus:after {
-          border: none;
```
Comment (xugao) [inline packages/experiments/src/components/Pagination/__snapshots__/Pagination.test.tsx.snap:1786]: some of this snapshot updates look unexpected

File: change/@fluentui-react-next-2020-07-30-18-07-42-a11y-menu.json (hunk 5 above)
Comment (tomi-msft) [reply change/@fluentui-react-next-2020-07-30-18-07-42-a11y-menu.json:4]: got it!

File: change/@fluentui-react-next-2020-07-30-18-07-42-a11y-menu.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "prerelease",
+  "comment": "update snapshots",
```
Comment (xugao) [inline change/@fluentui-react-next-2020-07-30-18-07-42-a11y-menu.json:3]:
```suggestion
  "comment": "update hc styling for Dropdown and ComboBox items to remove noisy borders",
```

File: change/@fluentui-react-next-2020-07-30-18-07-42-a11y-menu.json (hunk 5 above)
Comment (xugao) [reply change/@fluentui-react-next-2020-07-30-18-07-42-a11y-menu.json:4]: Your changes look great! pls apply the same in `react-next`

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts [hunk 8]
```diff
@@ -353,9 +349,28 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
...
+    dropdownItem: [
+      ...dropdownItemStyle,
+      itemSelectors(),
+      {
+        selectors: {
+          [HighContrastSelector]: {
+            border: 'none',
+          },
+        },
+      },
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts:361]: sorry for not noticing this earlier, i think this change should go in `itemSelector`

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts [hunk 9]
```diff
@@ -353,9 +349,28 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
...
-    dropdownItemDisabled: dropdownItemDisabled,
+    dropdownItemDisabled: [
+      dropdownItemDisabled,
+      {
+        selectors: {
+          [HighContrastSelector]: {
+            border: 'none',
+          },
+        },
+      },
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts:372]: same here, can go in `const dropdownItemDisabled` (defined above) and revert changes here

File: packages/react-next/src/components/Dropdown/Dropdown.styles.ts [hunk 10]
```diff
@@ -353,9 +349,28 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
     ],
     dropdownItemsWrapper: { selectors: { '&:focus': { outline: 0 } } },
     dropdownItems: [globalClassnames.dropdownItems, { display: 'block' }],
-    dropdownItem: [...dropdownItemStyle, itemSelectors()],
+    dropdownItem: [
```
Comment (xugao) [inline packages/react-next/src/components/Dropdown/Dropdown.styles.ts:352]: similar comments here as in OUFR

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts [hunk 11]
```diff
@@ -355,7 +355,7 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
     dropdownItems: [globalClassnames.dropdownItems, { display: 'block' }],
     dropdownItem: [...dropdownItemStyle, itemSelectors()],
     dropdownItemSelected: dropdownItemSelected,
-    dropdownItemDisabled: dropdownItemDisabled,
+    dropdownItemDisabled: [dropdownItemDisabled],
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts:358]: revert

Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts:358]:
```suggestion
    dropdownItemDisabled: dropdownItemDisabled,
```

File: packages/react-next/src/components/Dropdown/Dropdown.styles.ts [hunk 12]
```diff
@@ -355,7 +355,7 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
     dropdownItems: [globalClassnames.dropdownItems, { display: 'block' }],
     dropdownItem: [...dropdownItemStyle, itemSelectors()],
     dropdownItemSelected: dropdownItemSelected,
-    dropdownItemDisabled: dropdownItemDisabled,
+    dropdownItemDisabled: [dropdownItemDisabled],
```
Comment (xugao) [inline packages/react-next/src/components/Dropdown/Dropdown.styles.ts:358]:
```suggestion
    dropdownItemDisabled: dropdownItemDisabled,
```

**Pull request merged**